poetry/core/masonry/builders/sdist.py

```diff
@@ -87,11 +87,12 @@ def build(self, target_dir=None):  # type: (Path) -> Path
                 else:
                     tar.addfile(tar_info)  # Symlinks & ?
 
-            setup = self.build_setup()
-            tar_info = tarfile.TarInfo(pjoin(tar_dir, "setup.py"))
-            tar_info.size = len(setup)
-            tar_info.mtime = time.time()
-            tar.addfile(tar_info, BytesIO(setup))
+            if not self._poetry.get_project_config("disable-setup-file", False):
```

Comment:
How about having the config name to be `use-setup-file` so that it's easier to read here? Then users specify `use-setup-file = false` in their `pyproject.toml`.

Reply:
Might be better considering in the long run we want users to explicitly say they want to use the setup file. We could also consider something along the lines of `generate-`/`include-`/`provide-` iinstead too. Since, what we do is not generate a setup file to be inclided in the sdist.

Reply:
+1 for `generate-setup-file`